app/src/main/java/com/example/drawingApp/EditLayerAdapter.kt

```diff
@@ -0,0 +1,76 @@
+package com.example.drawingApp
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+
+class EditLayerAdapter(
+    private val data: List<LayerViewModel>,
+    /**
+     * called when the edit image is clicked in each view of the recycler view
+     * determines what will be performed when clicked
+     * currently allows the editing of the layer at that position of data
+     */
+    private val onEditClick: (LayerViewModel) -> Unit,
+    /**
+     * called when the delete image is clicked in each view of the recycler view
+     * determines what will be performed when clicked
+     * currently deletes the layer at that position of data
+     */
+    private val onDeleteClick: (LayerViewModel) -> Unit
+) : RecyclerView.Adapter<ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        return LayerViewHolder(
+            inflater.inflate(R.layout.edit_layer_dialog, parent, false),
+            onEditClick,
+            onDeleteClick
+        )
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(data[position])
+    }
+
+    override fun getItemCount(): Int = data.size
+
+}
+
+class LayerViewModel(val text: String)
+
+abstract class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    abstract fun bind(model: LayerViewModel)
+}
+
+class LayerViewHolder(
+    view: View,
+    onEdit: (LayerViewModel) -> Unit,
+    onDelete: (LayerViewModel) -> Unit
+) : ViewHolder(view) {
+    private val textView: TextView
+    private val editView: ImageView
+    private val deleteView: ImageView
+    private val editLayer = onEdit
+    private val deleteLayer = onDelete
```

Comment:
You're making private variables of things passed in the constructor. No need to do this, just add `val` to the variables in the constructor

Reply:
Implemented, but I'll be honost I'm not sure how this helps obfuscate this information away.

Reply:
implemented, at first I wasn't sure how this would help obfuscate the information, but I realized I could just remove the variables and just directly use the values passed to the constructor